src/Cluster/SlotRange.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Predis package.
+ *
+ * (c) 2009-2020 Daniele Alessandri
+ * (c) 2021-2025 Till Krüss
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Predis\Cluster;
+
+use Countable;
+use OutOfBoundsException;
+
+/**
+ * Represents a range of slots in a Redis cluster.
+ */
+class SlotRange implements Countable
+{
+    /**
+     * Maximum number of slots in a Redis cluster is 16384.
+     */
+    public const MAX_SLOTS = 0x3FFF;
+
+    /**
+     * Starting slot of the range.
+     *
+     * @var int
+     */
+    protected $start;
+
+    /**
+     * Ending slot of the range.
+     *
+     * @var int
+     */
+    protected $end;
+
+    /**
+     * Connection to the server hosting this slot range.
+     *
+     * @var string
+     */
+    protected $connection;
+
+    public function __construct(int $start, int $end, string $connection)
+    {
+        if (!static::isValidRange($start, $end)) {
+            throw new OutOfBoundsException("Invalid slot range $start-$end for `$connection`");
+        }
+        $this->start = $start;
+        $this->end = $end;
+        $this->connection = $connection;
+    }
+
+    /**
+     * Checks if a slot range is valid.
+     *
+     * @param int $first
+     * @param int $last
+     *
+     * @return bool
+     */
+    public static function isValidRange($first, $last)
+    {
+        return $first >= 0 && $first <= self::MAX_SLOTS && $last >= 0x0000 && $last <= self::MAX_SLOTS && $first <= $last;
```

Comment:
for consistency, this can be: 
```suggestion
        return $first >= 0x0000 && $first <= self::MAX_SLOTS && $last >= 0x0000 && $last <= self::MAX_SLOTS && $first <= $last;
```

Reply:
Good catch